l10n_it_fatturapa_out_rc/wizard/wizard_export_e_invoice.py

```diff
@@ -0,0 +1,127 @@
+from odoo.addons.l10n_it_fatturapa.bindings.fatturapa import (
+    IdFiscaleType,
+    AnagraficaType,
+    IndirizzoType
+)
+
+
+class WizardExportFatturapa(models.TransientModel):
+    _inherit = "wizard.export.fatturapa"
+
+    def exportInvoiceXML(
+        self, company, partner, invoice_ids, attach=False, context=None
+    ):
+        if context is None:
+            context = {}
+        invoices = self.env["account.invoice"].browse(invoice_ids)
+        invoices_with_rc = False
+        invoices_without_rc = False
+        for invoice in invoices:
+            if invoice.rc_purchase_invoice_id:
+                invoices_with_rc = True
+            else:
+                invoices_without_rc = True
+        if invoices_with_rc and invoices_without_rc:
+            raise UserError(_(
+                "Selected invoices are both with and without reverse charge. You "
+                "should selected a smaller set of invoices"))
+        rc_suppliers = invoices.mapped("rc_purchase_invoice_id.partner_id")
+        if len(rc_suppliers) > 1:
+            raise UserError(_(
+                "Selected reverse charge invoices have different suppliers. Please "
+                "select invoices with same supplier"))
+        if rc_suppliers:
+            context['rc_supplier'] = rc_suppliers[0]
+        return super(WizardExportFatturapa, self).exportInvoiceXML(
+            company, partner, invoice_ids, attach, context)
+
+    def _setDatiAnagraficiCedente(self, CedentePrestatore, company):
+        res = super(WizardExportFatturapa, self)._setDatiAnagraficiCedente(
+            CedentePrestatore, company)
+        if self.env.context.get("rc_supplier"):
+            partner = self.env.context["rc_supplier"]
+            CedentePrestatore.DatiAnagrafici.CodiceFiscale = None
+            if partner.vat:
+                CedentePrestatore.DatiAnagrafici.IdFiscaleIVA = IdFiscaleType(
+                    IdPaese=partner.vat[0:2], IdCodice=partner.vat[2:])
+            elif partner.country_id.code and partner.country_id.code != 'IT':
+                CedentePrestatore.DatiAnagrafici.IdFiscaleIVA = IdFiscaleType(
+                    IdPaese=partner.country_id.code, IdCodice='99999999999')
+            else:
+                raise UserError(
+                    _("Impossible to set IdFiscaleIVA for %s") % partner.display_name)
+            CedentePrestatore.DatiAnagrafici.Anagrafica = AnagraficaType(
+                Denominazione=partner.name)
+        return res
+
+    def _setSedeCedente(self, CedentePrestatore, company):
+        res = super(WizardExportFatturapa, self)._setSedeCedente(
+            CedentePrestatore, company)
+        if self.env.context.get("rc_supplier"):
+            partner = self.env.context["rc_supplier"]
+            if not partner.street:
+                raise UserError(
+                    _('Partner %s, Street is not set.') % partner.display_name)
+            if not partner.zip:
```

Comment:
Da qui in poi si dovrebbe riutilizzare la logica [qui](https://github.com/OCA/l10n-italy/blob/5e5fed251fdd1d692797e9951552a031a9db5f34/l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py#L489) visto che il CedentePrestatore potrebbe essere il fornitore estero e qui non essere valido ai fini della validazione completa.

Reply:
Oppure ho sbagliato qualche configurazione, nel cessionario/committente dovrei trovare la mia azienda ma trovo il fornitore.

Reply:
Ti confermo che nel committente dovresti trovare la tua azienda.

Ti consiglio di guardare cosa fanno i test per riprodurre la configurazione manualmente e  verificare che tutto torni.

Reply:
Mi par di capire che funziona solo se il reverse charge è impostato con il tipo `other`

Reply:
Esatto, perchè il partner dell'autofattura non deve essere il fornitore ma la tua azienda.
Vedi anche https://github.com/OCA/l10n-italy/issues/1937 "Implementazione"

Reply:
Il cedente è comunque il fornitore estero, quindi la segnalazione sul CAP e gli altri controlli sull'estero rimangono validi, anche impostando il tipo `other`,